pkg/networkmanager/firewall-client.js

```diff
@@ -253,15 +253,16 @@ function fetchServiceInfos(services) {
         let info;
         return firewalld_dbus.call('/org/fedoraproject/FirewallD1',
                                    'org.fedoraproject.FirewallD1',
-                                   'getServiceSettings', [service])
+                                   'getServiceSettings2', [service])
                 .then(reply => {
-                    const name = reply[0][1];
-                    const description = reply[0][2];
-                    const ports = reply[0][3];
+                    const obj = reply[0];
+                    const name = obj.short ? obj.short.v : "";
+                    const description = obj.description ? obj.description.v : "";
+                    const ports = obj.ports ? obj.ports.v : [];
```

Comment:
This would be another excellent use case for structured decomposition?

    .then(([{ short, description, ports }]) => { ... }

?

Reply:
I can squeeze that in.